frontend/vite.config.js

```diff
@@ -6,5 +6,29 @@ export default defineConfig({
   plugins: [react()],
   build: {
     outDir: '../backend/src/main/resources/static',
+  },
+  server: {
+    proxy: {
+      "/register": {
+        target: 'http://localhost:8080',
+        changeOrigin: true,
+        secure: false
+      },
+      "/login": {
+        target: 'http://localhost:8080',
+        changeOrigin: true,
+        secure: false
+      },
+      "/logout": {
```

Comment:
So for every endpoint, do I need to add it here? Let's say I create `/userId/survey` does it need to be added to this file?

Reply:
This was added to make our development process easier. Instead of needing to rebuild our backend everytime we make changes to the frontend, this proxies any requests made from the frontend to the backend. Therefore we don't need to keep rebuilding the backend. 

The problem is that we need it to proxy all requests except for "/". So we are allowlisting at this point all our routes. Instead, if we just proxy any requets to "/api" it will make this easier. All our controllers will then need to start with /api.